packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -127,12 +139,23 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     final int major = version?.major;
     final int minor = version?.minor;
     if (globals.platform.isMacOS) {
-      return globals.fs.path.join(
-        globals.fsUtils.homeDirPath,
-        'Library',
-        'Application Support',
-        'AndroidStudio$major.$minor',
-      );
+      /// plugin path of Android Studio has been changed after version 4.1.
+      if(double.parse('$major.$minor') >= 4.1) {
```

Comment:
Instead of concatenation these numbers parsing the string, check for the major and minor version separately:

`(if major >= 4 && minor >= 1) {`

Reply:
Got it!